scripts/save_embeddings.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import scanpy as sc
+import scIB
+import argparse
+import os
+import warnings
+warnings.filterwarnings('ignore')
+
+RESULT_TYPES = [
+    "full",  # reconstructed expression data
+    "embed", # embedded/latent space
+    "knn"    # corrected neighbourhood graph
+]
+
+if __name__=='__main__':
+    """
+    Save embeddings for all scenarios, methods and settings
+    """
+
+    parser = argparse.ArgumentParser(description='Save embeddings')
+
+    parser.add_argument('-i', '--input', required=True)
+    parser.add_argument('-o', '--outfile', required=True,
+                        help='Output coordinates CSV file')
+    parser.add_argument('-m', '--method', required=True, help='Name of method')
+    parser.add_argument('-r', '--result', required=True, choices=RESULT_TYPES,
+                        help='Type of result: full, embed, knn')
+    parser.add_argument('-b', '--batch_key', required=True, help='Key of batch')
+    parser.add_argument('-l', '--label_key', required=True,
+                        help='Key of annotated labels e.g. "cell_type"')
+
+    args = parser.parse_args()
+
+    outfile = args.outfile
+    method = args.method
+    result = args.result
+    setup = f'{method}_{result}'
+    batch_key = args.batch_key
+    label_key = args.label_key
+    adata = sc.read_h5ad(args.input)
+
+    print('Preparing dataset...')
+    if result == 'embed':
+        scIB.pp.reduce_data(adata, n_top_genes=None, neighbors=True,
+                            use_rep='X_emb', pca=False, umap=False)
+    elif result == 'full':
+        scIB.pp.reduce_data(adata, n_top_genes=2000, neighbors=True,
+                            use_rep='X_pca', pca=True, umap=False)
+
+    # Calculate embedding
+    if args.method.startswith('conos'):
+        print('Calculating graph embedding...')
+        sc.tl.draw_graph(adata)
+        # May be named differently depending on packages so get the last key
+        graph_key = list(adata.obsm.keys())[-1]
+        basis = graph_key.lstrip('X_')
+        label = basis.replace('draw_graph_', '').upper()
```

Comment:
Not quite sure I understand this bottom part. Why not just make it clear what the draw_graph_output key should be by stating the `key_added_ext` parameter or sth like that. You can then search for the extension you chose. Are we sure that the order of `adata.obsm.keys()` always conforms to the last key being the most recently introduced key?

Reply:
That is a much better idea. I will change it to do that.

Reply:
Ok. I did that. I have realised though that because ForceAtlas2 isn't in the environment it will always use the FR algorithm so I can probably just assume that is the label. Alternatively, we could add FA2 to the environment. Thoughts?

Reply:
I think we should probably add fa2 to the environment...

Reply:
But I do think this is now a good solution as is either way. Adding FA2 is a separate issue.